### PR TITLE
perf(modeling): remove use of slow Math.hypot

### PR DESCRIPTION
**Pull request by platypii**
It turns out that `Math.hypot` is much slower than `Math.sqrt`, as pointed out by @briansturgill in #1095

I liked using hypot for increased accuracy, and nicer to read. But since it is slower, I think it should go away.

I had to remove the tests for accuracy on very small, and very large numbers (1e200+).

### All Submissions:

* [x] Have you followed the guidelines in our Contributing document?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?
* [x] Does your submission pass tests?

### CODE REVIEW

**Review by hrgdavor — COMMENTED** [review]
looks good

**Comment by z3dev** [comment]
@platypii sorry. I'm not going for this without a good reason. For places where performance matters then the larger function can opt for using sqrt(). I think we should open an issue with gl matrix, and ask why that library uses hypot()

**Comment by platypii** [comment]
@z3dev that's fair. I'll see if I can measure the speed difference in real models.

**Comment by platypii** [comment]
Changing from `hypot` to `sqrt` makes a significant 30%+ speed up on unions. I think it's worth considering the change.

|  | hypot | no hypot | improvement |
| --- | --- | --- | --- |
| disjoint (x=4) | 480ms | 480ms | 0% |
| touching (x=2) | 8500ms | 5800ms | 31.8% |
| overlap (x=1) | 8900ms | 5850ms | 34.3% |
| inside (x=0) | 7300ms | 4800ms | 34.2% |

<details>
<summary>Test script</summary>

```js
const jscad = require('@jscad/modeling')
const { sphere } = jscad.primitives
const { union } = jscad.booleans

const main = () => {
  const segments = 100
  const x = 1
  let s1 = sphere({ segments })
  let s2 = sphere({ segments, center: [x, 0, 0] })
  const startTime = performance.now()
  const out = union(s1, s2)
  console.log(`Union time ${performance.now() - startTime} ms`)
  return out
}

module.exports = { main }
```
</details>

**Comment by hrgdavor** [comment]
that is massive, definitely worth changing

**Comment by z3dev** [comment]
> Changing from `hypot` to `sqrt` makes a significant 30%+ speed up on unions. I think it's worth considering the change.
> 
> hypot	sqrt	improvement
> disjoint (x=4)	480ms	480ms	0%
> touching (x=2)	8500ms	5800ms	31.8%
> overlap (x=1)	8900ms	5850ms	34.3%
> inside (x=0)	7300ms	4800ms	34.2%
> Test script

I'm not saying that the replacement shouldn't be done. I'm saying that we should be selective about the replacement. Booleans may benefit but we don't want to create other issues.

We should do some research, and ask the gl matrix geeks. And try to understand the math completely.

**Comment by platypii** [comment]
https://github.com/toji/gl-matrix/issues/446

**Comment by z3dev** [comment]
FYI, here's the js-core implementation of hypot()

https://github.com/zloirock/core-js/blob/master/packages/core-js/modules/es.math.hypot.js

**Comment by platypii** [comment]
The core-js polyfill implementation is somehow even slower than Math.hypot despite using sqrt. :vomiting_face: 

| environment | hypot | sqrt | core-js |
|-------------|-------|------|-------|
| chrome      | 235ms | 12ms | 328ms |
| firefox     | 64ms  | 12ms | 160ms |
| safari      | 390ms | 74ms | 1830ms |
| node        | 248ms | 11ms | 370ms |

File: packages/modeling/src/maths/mat4/rotate.js [hunk 1]
```diff
@@ -14,14 +15,14 @@ const copy = require('./copy')
  */
 const rotate = (out, matrix, radians, axis) => {
   let [x, y, z] = axis
-  let len = Math.hypot(x, y, z)
+  const lengthSquared = x * x + y * y + z * z
 
-  if (Math.abs(len) < 0.000001) {
+  if (Math.abs(lengthSquared) < EPSILON) {
```
Comment (z3dev) [inline packages/modeling/src/maths/mat4/rotate.js:20]: I suggest that we use the EPS for normals, then we know that this function works for calculations with normals.

File: packages/modeling/src/maths/mat4/fromRotation.js [hunk 2]
```diff
@@ -21,14 +21,14 @@ const { EPSILON } = require('./constants')
  */
 const fromRotation = (out, rad, axis) => {
   let [x, y, z] = axis
-  let len = Math.hypot(x, y, z)
+  const lengthSquared = x * x + y * y + z * z
 
-  if (Math.abs(len) < EPSILON) {
+  if (Math.abs(lengthSquared) < EPSILON) {
```
Comment (z3dev) [inline packages/modeling/src/maths/mat4/fromRotation.js:26]: Let's use the EPS for normals.

**Review by z3dev — CHANGES_REQUESTED** [review]

**Review by z3dev — APPROVED** [review]
@platypii thanks again. Let's see if the general user community feels any performance improvement.

**Pull request merged**